addons/stock/static/src/js/forecast_widget.js

```diff
@@ -3,22 +3,28 @@ odoo.define('stock.forecast_widget', function (require) {
 
 const AbstractField = require('web.AbstractField');
 const fieldRegistry = require('web.field_registry');
+const field_utils = require('web.field_utils');
 const core = require('web.core');
 const QWeb = core.qweb;
 
 const ForecastWidgetField = AbstractField.extend({
-    supportedFieldTypes: ['char'],
+    supportedFieldTypes: ['float'],
 
     _render: function () {
-        const forecastData = JSON.parse(this.value);
-        if (!forecastData) {
-            this.$el.html('');
-        } else {
-            // Don't display the graph button for non-storable products or if was explicited.
-            forecastData.hideReportButton = this.recordData.product_type !== 'product';
-            this.$el.html(QWeb.render('stock.forecastWidget', forecastData));
-            this.$el.on('click', this._onOpenReport.bind(this));
+        var data = Object.assign({}, this.record.data, {
+            forecast_availability_str: field_utils.format.float(this.record.data.forecast_availability, this.record.fields.forecast_availability, this.nodeOptions),
+            reserved_availability_str: field_utils.format.float(this.record.data.reserved_availability, this.record.fields.reserved_availability, this.nodeOptions),
+            forecast_expected_date_str: field_utils.format.date(this.record.data.forecast_expected_date, this.record.fields.forecast_expected_date),
+        });
+        if (this.record.data.forecast_expected_date && this.record.data.date_deadline) {
+            data.forecast_is_late = this.record.data.forecast_expected_date > this.record.data.date_deadline;
```

Comment:
```suggestion
        const data = Object.assign({}, this.record.data, {
            forecast_availability_str: field_utils.format.float(
                this.record.data.forecast_availability,
                this.record.fields.forecast_availability,
                this.nodeOptions
            ),
            reserved_availability_str: field_utils.format.float(
                this.record.data.reserved_availability,
                this.record.fields.reserved_availability,
                this.nodeOptions
            ),
            forecast_expected_date_str: field_utils.format.date(
                this.record.data.forecast_expected_date,
                this.record.fields.forecast_expected_date
            ),
        });
        if (data.forecast_expected_date && data.date_deadline) {
            data.forecast_is_late = data.forecast_expected_date > data.date_deadline;
```
Line break for better readability, and `data` instead of `this.record.data` as data is a copy of that.